packages/amplify-e2e-core/src/init/amplifyPush.ts

```diff
@@ -36,6 +36,39 @@ export function amplifyPushForce(cwd: string, testingWithLatestCodebase: boolean
   });
 }
 
+/**
+ *
+ * @param cwd
+ * @param stackIdx
+ * @param testingWithLatestCodebase
+ * @returns
+ * Used to stop an iterative deployment
+ */
+export function cancelIterativeAmplifyPush(
+  cwd: string,
+  stackIdx: { current: number; max: number },
+  testingWithLatestCodebase: boolean = false,
+): Promise<void> {
+  return new Promise((resolve, reject) => {
+    spawn(getCLIPath(testingWithLatestCodebase), ['push'], { cwd, stripColors: true, noOutputTimeout: pushTimeoutMS })
+      .wait('Are you sure you want to continue?')
+      .sendLine('y')
+      .wait(`Deploying stack (${stackIdx.current} of ${stackIdx.max})`)
+      .wait(/.*UPDATE_IN_PROGRESS*/)
+      .sendLine('\x03')
```

Comment:
I'm assuming this is the code for ctrl+C? Maybe define it in a constant to make that more obvious